src/jit/codegenlegacy.cpp

```diff
@@ -11364,8 +11364,6 @@ void CodeGen::genCodeForTreeSmpOpAsg(GenTreePtr tree)
 
             if (VarSetOps::IsMember(compiler, genUpdateLiveSetForward(op2), varDsc->lvVarIndex))
             {
-                noway_assert(compiler->opts.compDbgCode);
```

Comment:
I have no idea whether it is new or not.
My thought was like why we are checking for generating 'debug-safe' code under condition that local var alive inside expression. It has no sense for me so I decided to delete it.